orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py

```diff
@@ -0,0 +1,1384 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+import pickle
+from numpy.testing import assert_allclose
+import numpy as np
+import argparse
+import glob
+
+import torch
+import torch.distributed as dist
+
+import onnxruntime
+from onnxruntime.training import amp, checkpoint, optim, orttrainer
+from _test_helpers import distributed_setup, load_evaluate_extract_state_dict, split_state_dict, split_name, aggregate_states
+
+def test_load_from_single_node_full_precision_into_single_node_full_precision(device = 'cuda', checkpoint_dir = 'checkpoint_dir/single_node/full_precision/'):
+    opts = {'device' : {'id' : device},
+            'debug' : {'deterministic_compute': True}}
+    
+    # extract state dictionaries to compare
+    state_dict_post_checkpoint, model = load_evaluate_extract_state_dict(device, orttrainer.ORTTrainerOptions(opts), checkpoint_dir)
+
+    state = pickle.load(open(os.path.join(checkpoint_dir, 'state_dict.pkl'), 'rb'))
+    state_dict_pre_checkpoint = state['state_dict']
+
+    # compare all states
+    for key, value in state_dict_pre_checkpoint.items():
+        assert_allclose(value, state_dict_post_checkpoint[key])
+    
+    # load state into pytorch and compare
+    model.load_state_dict(state_dict_pre_checkpoint, strict=False)
+    state_dict_pytorch = model.state_dict()
+    for key, value in state_dict_pytorch.items():
+        assert_allclose(value, state_dict_pre_checkpoint[key])
+
+def test_load_from_single_node_mixed_precision_into_single_node_full_precision(device = 'cuda', checkpoint_dir = 'checkpoint_dir/single_node/mixed_precision/'):
+    opts = {'device' : {'id' : device},
+            'debug' : {'deterministic_compute': True}}
+    
+    # extract state dictionaries to compare
+    state_dict_post_checkpoint, model = load_evaluate_extract_state_dict(device, orttrainer.ORTTrainerOptions(opts), checkpoint_dir)
+
+    state = pickle.load(open(os.path.join(checkpoint_dir, 'state_dict.pkl'), 'rb'))
+    state_dict_pre_checkpoint = state['state_dict']
+
+    # compare all states
+    for key, value in state_dict_pre_checkpoint.items():
+        if key.endswith('_fp16'):
+            full_precision_key = key[:-5]
+            assert full_precision_key in state_dict_post_checkpoint
+            assert_allclose(value, state_dict_post_checkpoint[full_precision_key], atol=1e-3)
```

Comment:
We should probably define a global tolerance for fp16 and another for fp32 and use the same across all tests
Previous experience showed that people will fix specific tests and CI will fail many times before all tests are fixed, one by one